### ISSUE TITLE
show more than one data block in the graph

### ISSUE THREAD

**Issue by yoelk** [yoelk/instrumentino2#37]
In commit [8bacc9f] (https://github.com/yoelk/instrumentino2/commit/8bacc9f368e9d6b4b16cb3a922d77772dad74e31) I added the ability to disconnect from a controller and reconnect. The data from each connected session is saved in data blocks.

Currently the graph can only show one block, and if two blocks are in sight, it only shows the last one.
We need to add the possibility to see more than one block, and somehow separating them on the screen.
This will probably mean having a plot per each graph, and drawing a vertical line between blocks to distinguish them from each other. They can't overlap in time.

**Comment by snarfums** [comment]
I think you're going to have 1 plot per each graph. Note: The garden.graph
only suports 8 plots (from the__init__.py source code docs):
"The graph uses a stencil view to clip the plots to the graph display area. As
with the stencil graphics instructions, you cannot stack more than 8
stencil-aware
widgets"

**Comment by yoelk** [comment]
Yes, you're probably right. So I hope there's a possibility to have a
broken graph, with empty areas where nothing is plotted.
Can you explain me what this stencil graphics actually mean?
Currently I'm providing the graph view each time a new graph, that fits the
current data I want to display.
Is this the right way? Or can I use this stencil graphics somehow to make
things easier?

**Comment by snarfums** [comment]
Q: Can you explain me what this stencil graphics actually mean?
A: Nope :) I have no idea.  Without reading up on it, It *seems* like it's like an HTML "canvas", which is an area to draw things/place graphics, but I don't know why it has such a low limitation on the number of things you can place in/on it.  Maybe there's some sort of override method (or one that could  be made) and the limit is just using the default stencil settings.

Q: "Currently I'm providing the graph view each time a new graph, that fits the current data I want to display. Is this the right way?"
A: I also have no idea :)  When I played around with it I did the same thing (or followed the same examples). I woudl think it "better" to just be able to add/push a new data point onto one end of the graph and (maybe) pop/remove one at the other, and then force a redraw.  reloading all the data into a graph view seems...wrong somehow... but I don't know of an alternate method right now.

Q: Or can I use this stencil graphics somehow to make things easier?
A: Maybe?  The garden.graph code supports a lot of different graph types. Maybe building a custom one on top of the "stencil" woudl work better?  Alternatively, and you mentioned this in some other discussions, maybe we look into an alternate graphing mechanism in order to be able to easier support data analysis?  Given the difficulties, it might be worth doing this after everything is "working" as this (anything related to graphing/data analysis) seems like a bit of work.

**Comment by yoelk** [comment]
Yes, I agree. I don't plan on touching data analysis anytime soon.
But I would like to improve performance a little. When you try having
several analog signals at high sampling rates the graph isn't so smooth.
Maybe I need to change the data packet rate to 20 instead of 10 per
second...
